File: exercises/js/day2-step1-setup-account.js
// ============================================================
// Day 2 - Step 1 Exercise: Setup Account (Enable Default Ripple)
// Learn: How to configure account flags on the XRP Ledger.
// Default Ripple is required before issuing tokens.
// ============================================================

const xrpl = require('xrpl');
const fs = require('fs');

const TESTNET_URL = 'wss://s.altnet.rippletest.net:51233';

async function main() {
  const { wallets } = JSON.parse(fs.readFileSync('../../wallets.json', 'utf-8'));
  const issuerWallet = xrpl.Wallet.fromSeed(wallets[0].seed);

  const client = new xrpl.Client(TESTNET_URL);
  await client.connect();
  console.log('Connected to XRPL Testnet');

  // ╔══════════════════════════════════════════════════════════════════╗
  // ║ TODO 1: Submit an AccountSet transaction to enable Default       ║
  // ║         Ripple on the issuer account                             ║
  // ║                                                                  ║
  // ║ HINT: {                                                          ║
  // ║   TransactionType: 'AccountSet',                                 ║
  // ║   Account: issuerWallet.address,                                 ║
  // ║   SetFlag: xrpl.AccountSetAsfFlags.asfDefaultRipple              ║
  // ║ }                                                                ║
  // ║                                                                  ║
  // ║ Submit: client.submitAndWait(tx, { wallet: issuerWallet })       ║
  // ╚══════════════════════════════════════════════════════════════════╝
  // YOUR CODE HERE

  console.log('Default Ripple enabled on issuer account!');
  console.log(`\nExplorer: https://testnet.xrpl.org/accounts/${issuerWallet.address}`);
  await client.disconnect();
}

main().catch(console.error);

File: solutions/python/day2_step1_setup_account.py
"""
Day 2 Step 1 Solution: Setup Account
=====================================
Enable Default Ripple flag on the issuer account.
"""

import json
from xrpl.clients import WebsocketClient
from xrpl.wallet import Wallet
from xrpl.models.transactions import AccountSet
from xrpl.transaction import submit_and_wait

TESTNET_URL = 'wss://s.altnet.rippletest.net:51233'

# Load wallets
with open('../../wallets.json') as f:
    data = json.load(f)
    wallets = data['wallets']

issuer = Wallet.from_seed(wallets[0]['seed'])
print(f"Issuer: {issuer.address}")

with WebsocketClient(TESTNET_URL) as client:

    # Enable Default Ripple
    account_set_tx = AccountSet(
        account=issuer.address,
        set_flag=8
    )
    result = submit_and_wait(account_set_tx, client, issuer)
    print(f"Result: {result.result['meta']['TransactionResult']}")

# Explorer: https://testnet.xrpl.org/accounts/{address}
print(f"\nExplorer: https://testnet.xrpl.org/accounts/{issuer.address}")
